contracts/governance/extensions/GovernorTimelock.sol

```diff
@@ -0,0 +1,201 @@
+
+        return eta;
+    }
+
+    /**
+     * @dev See {IGovernor-_execute}
+     */
+    function _execute(
+        uint256 proposalId,
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        bytes32 /*descriptionHash*/
+    ) internal virtual override {
+        ExecutionDetail[] storage details = _executionDetails[proposalId];
+        ExecutionDetail memory detail;
+
+        for (uint256 i = 0; i < targets.length; ++i) {
+            detail = details[i];
+            if (detail.manager != address(0)) {
+                IAccessManager(detail.manager).relay{value: values[i]}(targets[i], calldatas[i]);
+            } else {
+                (bool success, bytes memory returndata) = targets[i].call{value: values[i]}(calldatas[i]);
+                Address.verifyCallResult(success, returndata);
+            }
+        }
+
+        delete _executionDetails[proposalId];
+    }
+
+    /**
+     * @dev See {IGovernor-_cancel}
+     */
+    function _cancel(
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory calldatas,
+        bytes32 descriptionHash
+    ) internal virtual override returns (uint256) {
+        uint256 proposalId = super._cancel(targets, values, calldatas, descriptionHash);
+
+        ExecutionDetail[] storage details = _executionDetails[proposalId];
+        ExecutionDetail memory detail;
+
+        for (uint256 i = 0; i < targets.length; ++i) {
+            detail = details[i];
+            if (detail.manager != address(0)) {
+                IAccessManager(detail.manager).cancel(address(this), targets[i], calldatas[i]);
+            }
+        }
+
+        delete _executionDetails[proposalId];
+
+        return proposalId;
+    }
+
+    /**
+     * @dev Default delay to apply to function calls that are not (scheduled and) executed through an AccessManager.
+     */
+    function _defaultDelay() internal view virtual returns (uint32) {
+        return 0;
+    }
+
+    /**
+     * @dev Check if the execution of a call needs to be performed through an AccessManager and what delay should be
+     * applied to this call.
+     *
+     * Returns { manager: address(0), delay: _defaultDelay() } if:
+     * - target does not have code
+     * - target does not implement IManaged
+     * - calling canCall on the target's manager returns a 0 delay
+     * - calling canCall on the target's manager reverts
+     * Otherwise (calling canCall on the target's manager returns a non 0 delay), return the address of the
+     * AccessManager to use, and the delay for this call.
+     */
+    function _detectExecutionDetails(address target, bytes4 selector) private view returns (ExecutionDetail memory) {
+        // If target is not a contract, skip
+        if (target.code.length > 0) {
+            // Try to fetch authority. If revert, skip
+            try IManaged(target).authority() returns (address authority) {
```

Comment:
`authority()` or a clashing signature may be defined but not return enough data, and this would cause the whole thing to revert. I don't think we can use `try` here, unless we do the decoding manually something like:

```solidity
            try IManaged(target).authority() {
                uint256 returnDataSize;
                address authority;
                assembly {
                    returnDataSize := returndatasize()
                    returndatacopy(0, 0, 32)
                    authority := mload(0)
                }
```

Reply:
done.